### PR TITLE
feat: ✨ Add stop time functionality to Proofs Dealer pallet

### PR DESCRIPTION
**Pull request by ffarall**

### CODE REVIEW

**Review by snowmead — APPROVED** [review]
Beast!

File: pallets/proofs-dealer/src/utils.rs [hunk 1]
```diff
@@ -565,6 +557,100 @@ where
...
+                // If the weight left in the previous block is greater or equal than the headroom, for both proof size or ref time,
+                // we consider the previous block to be NOT full and count it as such.
+                if weight_left_in_prev_block.ref_time()
+                    >= T::BlockFullnessHeadroom::get().ref_time()
+                    && weight_left_in_prev_block.proof_size()
+                        >= T::BlockFullnessHeadroom::get().proof_size()
+                {
+                    // Increment the counter of blocks that are not full.
+                    NotFullBlocksCount::<T>::mutate(|n| *n = n.saturating_add(1u32.into()));
+                    weight.consume(T::DbWeight::get().reads_writes(1, 1));
```
[the remark this answers is not in the document]
Comment (ffarall) [reply pallets/proofs-dealer/src/utils.rs:596]: You're 100% right. Good catch.

[the remark this answers is not in the document]
Comment (ffarall) [reply pallets/proofs-dealer/src/utils.rs:596]: The thing though, is that in the second write to this variable I don't know if the first write was executed.

File: test/util/bspNet/consts.ts [hunk 2]
```diff
@@ -85,3 +85,8 @@ export const CAPACITY = {
 
 export const U32_MAX = (BigInt(1) << BigInt(32)) - BigInt(1);
 export const MAX_STORAGE_CAPACITY = CAPACITY[1024] * 4n - 1n;
+
+export const REMARK_WEIGHT_REF_TIME = 127_121_340;
+export const REMARK_WEIGHT_PROOF_SIZE = 142;
+export const TRANSFER_WEIGHT_REF_TIME = 297_297_000;
+export const TRANSFER_WEIGHT_PROOF_SIZE = 308;
```
[the remark this answers is not in the document]
Comment (ffarall) [reply test/util/bspNet/consts.ts:92]: I got them by measuring the execution when running a block with that extrinsic

**Pull request merged**